tinygrad/mlops.py

```diff
@@ -89,7 +89,8 @@ def backward(self, grad_output:LazyBuffer) -> LazyBuffer:
 
 class Less(Function):
   def forward(self, x:LazyBuffer, y:LazyBuffer) -> LazyBuffer:
-    return x.e(BinaryOps.CMPLT, y)
+    vars_dtype = least_upper_dtype(x.dtype, y.dtype)
+    return x.cast(vars_dtype).e(BinaryOps.CMPLT, y.cast(vars_dtype))
```

Comment:
i think we convert output to dtypes.bool here, not lazy. you can include this https://github.com/chenyuxyz/tinygrad/pull/55

Reply:
linearizer always interprets CMPLT as bool that's why we need to cast here https://github.com/tinygrad/tinygrad/pull/2782

Reply:
but it doesn't logically make any sense for cmplt to be anything but bool? I think we still need
https://github.com/tinygrad/tinygrad/pull/2800/commits/cd2bb4d7520308f7291e064078cbba248805b12f
otherwise the lazyop info would promote the dtypes

Reply:
the solution might be LazyOp stores the output dtype, and not recompute the dtype in get_lazy_op_info

Reply:
a part of what I did is this:
https://github.com/tinygrad/tinygrad/pull/2800/files#diff-566a7254fa321896c55ed04f6d3eb9cf38ac84ee952b4e44a107c006cf886b16R239